cohort-engine/src/main/java/com/ibm/cohort/engine/measure/seed/MeasureEvaluationSeeder.java

```diff
@@ -0,0 +1,175 @@
+import org.opencds.cqf.cql.engine.data.DataProvider;
+import org.opencds.cqf.cql.engine.debug.DebugMap;
+import org.opencds.cqf.cql.engine.execution.Context;
+import org.opencds.cqf.cql.engine.execution.LibraryLoader;
+import org.opencds.cqf.cql.engine.runtime.DateTime;
+import org.opencds.cqf.cql.engine.runtime.Interval;
+import org.opencds.cqf.cql.engine.terminology.TerminologyProvider;
+import org.opencds.cqf.r4.helpers.LibraryHelper;
+
+public class MeasureEvaluationSeeder {
+
+	private final EvaluationProviderFactory providerFactory;
+	private final LibraryLoader libraryLoader;
+	private final LibraryResolutionProvider<Library> libraryResourceProvider;
+
+	private boolean disableExpressionCaching;
+	private boolean debugMode = true;
+	private String terminologyProviderSource;
+	private String terminologyProviderUser;
+	private String terminologyProviderPassword;
+
+	public MeasureEvaluationSeeder(
+			EvaluationProviderFactory providerFactory,
+			LibraryLoader libraryLoader,
+			LibraryResolutionProvider<Library> libraryResourceProvider) {
+		this.providerFactory = providerFactory;
+		this.libraryLoader = libraryLoader;
+		this.libraryResourceProvider = libraryResourceProvider;
+	}
+
+	public MeasureEvaluationSeeder disableDebugLogging() {
+		this.debugMode = false;
+
+		return this;
+	}
+
+	public MeasureEvaluationSeeder disableExpressionCaching() {
+		this.disableExpressionCaching = true;
+
+		return this;
+	}
+
+
+	public MeasureEvaluationSeeder withTerminologyProvider(String source, String user, String password) {
+		this.terminologyProviderSource = source;
+		this.terminologyProviderUser = user;
+		this.terminologyProviderPassword = password;
+
+		return this;
+	}
+
+
+	public IMeasureEvaluationSeed create(Measure measure, String periodStart, String periodEnd, String productLine) {
+		LibraryHelper.loadLibraries(measure, this.libraryLoader, this.libraryResourceProvider);
+
+		// resolve primary library
+		org.cqframework.cql.elm.execution.Library library = LibraryHelper.resolvePrimaryLibrary(measure, libraryLoader, this.libraryResourceProvider);
+
+		List<Triple<String, String, String>> usingDefs = UsingHelper.getUsingUrlAndVersion(library.getUsings());
+
+		if (usingDefs.size() > 1) {
+			throw new IllegalArgumentException(
+					"Evaluation of Measure using multiple Models is not supported at this time.");
+		}
+
+
+		TerminologyProvider terminologyProvider = createTerminologyProvider(usingDefs);
+		LinkedHashMap<Triple<String, String, String>, DataProvider> dataProviders = createDataProviders(usingDefs, terminologyProvider);
+		Interval measurementPeriod = createMeasurePeriod(periodStart, periodEnd);
+		Context context = createContext(library, dataProviders, terminologyProvider, measurementPeriod, productLine);
+
+		// todo: [daniel.kim] this only returns the last seen data provider
+		// is this a bug?
+		List<Map.Entry<Triple<String, String, String>, DataProvider>> dataProviderList = new ArrayList<>(dataProviders.entrySet());
```

Comment:
The data provider that is stored in a member field and used downstream is used only once and that is to resolve the Patient class by patient ID. This _could_ go wrong if you had "using FHIR" and then later "using QDM" and the patient retrieval had to be FHIR. In reality, this is never going to happen. It would be a very strange use case to mix data providers. I think this is fine for now.